go/engine/resolve_identify2.go

```diff
@@ -141,3 +156,49 @@ func (e *ResolveThenIdentify2) GetProofSet() *libkb.ProofSet {
 	}
 	return e.i2eng.GetProofSet()
 }
+
+// ResolveAndCheck takes as input a name (joe), social assertion (joe@twitter)
+// or compound assertion (joe+joe@twitter+3883883773222@pgp) and invokes
+// the whole resolve/identify machinery. That is, it performs a server-trusted
+// resolution of the name to a UID, then does an ID on the UID to make sure all
+// assertions are met. Pass into it a MetaContext without any UIs set, since it
+// is meant to run without any UI interaction. Also note that tracker statements
+// are *not* taken into account. The identify is run as if the user is logged out.
+// The output, on success, is a populated UserPlusKeysV2.
+func ResolveAndCheck(m libkb.MetaContext, s string) (ret keybase1.UserPlusKeysV2, err error) {
```

Comment:
This is a mix of things that are part of the function's contract and things that aren't. How about splitting it up like this?

```
// ResolveAndCheck takes as input a name (joe), social assertion (joe@twitter)
// or compound assertion (joe+joe@twitter+3883883773222@pgp) and resolves
// it to a user, verifying the result.
// Pass into it a MetaContext without any UIs set, since it
// is meant to run without any UI interaction. Also note that tracker statements
// are *not* taken into account.
// The output, on success, is a populated UserPlusKeysV2.
func ResolveAndCheck(...
    // Invokes the whole resolve/identify machinery. That is, it performs a server-trusted
    // resolution of the name to a UID, then does an ID on the UID to make sure all
    // assertions are met. The identify is run as if the user is logged out.
```

Also what caching is part of the contract? It will never return old results?

Reply:
yeah, just upak caching, and it forces fresh, i believe...